sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/FilterEstimation.scala

```diff
@@ -311,6 +311,16 @@ case class FilterEstimation(plan: Filter) extends Logging {
       logDebug("[CBO] No statistics for " + attr)
       return None
     }
+
+    attr.dataType match {
+      case _: NumericType | DateType | TimestampType | BooleanType =>
+        if (!colStatsMap.hasMinMaxStats(attr)) {
+          logDebug(s"[CBO] No statistics for $attr")
+          return None
+        }
+      case _ =>
+    }
+
     val colStat = colStatsMap(attr)
 
     // decide if the value is in [min, max] of the column.
```

Comment:
If the NumericType column does not store min/max properties, the statsInterval will be NullValueInterval(), then the evaluateEquality() method will return Some(0.0), not None。